doc/sphinx/proof-engine/tactics.rst

```diff
@@ -549,8 +549,12 @@ one or more of its hypotheses.
    which is equivalent to `in * |- *`.  Use `* |-` to select all occurrences
    in all hypotheses.
 
-   Tactics that select a specific hypothesis H to apply to other hypotheses,
-   such as :tacn:`rewrite` `H in * |-`, won't apply H to itself.
+   When rewriting in hypotheses, they must not appear in the term to
+   rewrite. For instance `rewrite H in H` is an error. If an
+   hypothesis appears only through an evar instance, the evar will be
+   restricted to avoid the dependency. With `rewrite term in *`,
+   hypotheses on which the dependency cannot be avoided are skipped,
+   for instance `rewrite H in *` skips rewriting in `H`.
```

Comment:
```suggestion
   Hypotheses can't be used to rewrite themselves.  For example, `rewrite H in H` is an error.
   For :tacn:`rewrite` `H in * |-` or `rewrite` `H in *`, `H` will not be applied to itself.
   Similarly, if `H` appears in an evar used in one of the selected hypotheses, it will not be rewritten.
```

If I've not oversimplified or mistated...  Not sure exactly what you mean by "the evar will be restricted to avoid the dependency" if it's not covered by "Similarly, if `H` appears through an evar, it will not be rewritten."

The original statement is still true and you're just being more explicit?